joblib/test/test_parallel.py

```diff
@@ -864,6 +867,20 @@ def test_warning_about_timeout_not_supported_by_backend():
         "will not be used.")
 
 
+def set_list_value(input_list, index, value):
+    input_list[index] = value
+    return value
+
+
+def test_parallel_return_generator():
```

Comment:
Maybe here a small comment to say that this test is inserting values in a list in an order in sequential computing, and checking that that order is respected.